scripts/autogen

```diff
@@ -799,6 +799,62 @@ def gen_hol_light_asm(dry_run=False):
     gen_hol_light_asm_file("ntt_opt.S", "mlkem_ntt.S", dry_run=dry_run)
     gen_hol_light_asm_file("intt_opt.S", "mlkem_intt.S", dry_run=dry_run)
 
+def gen_header_guard(header_file, dry_run=False):
+    # Checks that the first non-comment line in the header file
+    # if of the form `#if defined(MLKEM_NATIVE_{full filename}_H)`.
+    with open(header_file, "r") as f:
+        content = f.read().split('\n')
+
+    # Use full filename as the header guard, with '/' and '.' replaced by '_'
+    guard_name = header_file.removeprefix("mlkem/").replace("/","_").replace(".","_").upper()
+    guard_name = "MLKEM_NATIVE_" + guard_name
+
+    def gen_copyright():
+        yield "/*"
+        yield " * Copyright (c) 2024 The mlkem-native project authors"
```

Comment:
Should we change this to 2024-2025? or something like that?

Reply:
Good point, will add this

Reply:
Changed this everywhere... hopefully